### ISSUE TITLE
`import_playbook` with a `.yml` extension is not resolved

### ISSUE THREAD

**Issue by bandit420** [ansible/ansible-lint#5189]
##### Summary

`ansible.builtin.import_playbook: community.molecule.validate.yml` fails with
`ERROR Failed to find community.molecule.validate.yml playbook.` The same reference without the extension
(`community.molecule.validate`) resolves. ansible-core supports both spellings, so ansible-lint should too.

Reproduced with 26.8.0 and with `main` (26.8.1.dev28).

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```bash
$ ansible-lint --version
ansible-lint 26.8.0 using ansible-core:2.21.4 ansible-compat:26.8.0 ruamel-yaml:0.19.1 ruamel-yaml-clib:None
```

- OS: CachyOS (Arch), x86_64, Python 3.14.7
- ansible installation method: pip (venv, managed with uv)
- ansible-lint installation method: pip (venv, managed with uv)

##### STEPS TO REPRODUCE

```bash
$ ansible-galaxy collection install community.molecule

$ printf -- '---\n- name: Ext\n  ansible.builtin.import_playbook: community.molecule.validate.yml\n' > ext.yml
$ printf -- '---\n- name: Noext\n  ansible.builtin.import_playbook: community.molecule.validate\n' > noext.yml

$ ansible-playbook --syntax-check ext.yml
playbook: ext.yml

$ ansible-lint --offline ext.yml
ERROR    Failed to find community.molecule.validate.yml playbook.

$ ansible-lint --offline noext.yml
```

##### Desired Behavior

Both spellings resolve, as in ansible-core, to
`<collections_path>/ansible_collections/community/molecule/playbooks/validate.yml`, and no
`Failed to find … playbook.` line is logged.

##### Actual Behavior

```bash
$ ansible-lint --offline ext.yml
ERROR    Failed to find community.molecule.validate.yml playbook.

Passed: 0 failure(s), 0 warning(s) in 1 files processed of 1 encountered. Last profile that met the validation criteria was 'production'.

$ ansible-lint --offline noext.yml

Passed: 0 failure(s), 0 warning(s) in 1 files processed of 1 encountered. Last profile that met the validation criteria was 'production'.

$ ansible-playbook --syntax-check ext.yml
playbook: ext.yml
```

- The last command succeeds: the reference is valid for ansible-core.
- Both runs exit 0. The error is only logged, and the imported playbook is not linted (it is not registered as a foreign
  playbook either), so the line is the only symptom.

##### Root cause

`parse_fqcn()` (utils.py) splits the reference on every dot, so the extension becomes a path segment:
`parse_fqcn("community.molecule.validate.yml")` → `("community", "molecule", "validate", "yml")`. The candidate paths are
then built by appending another extension to the last segment, giving
`…/community/molecule/playbooks/validate/yml.yml` and `…/validate/yml.yaml`, which cannot exist.

##### Suggested fix

Strip a trailing `.yml` / `.yaml` from the playbook resource before building the candidate paths, mirroring
`AnsibleCollectionRef.from_fqcr()` in ansible-core, which special-cases `ref_type == "playbook"` for exactly that.
